File: 2461-maximum-sum-of-distinct-subarrays-with-length-k/2461-maximum-sum-of-distinct-subarrays-with-length-k.java
class Solution {
    public long maximumSubarraySum(int[] nums, int k) {
        HashMap<Integer, Integer> mpp = new HashMap<>();
        int l=0;
        long sum = 0;
        long maxsum = 0;

        for(int i=0;i<nums.length;i++){
            sum += nums[i];
            mpp.put(nums[i], mpp.getOrDefault(nums[i],0)+1);
            if(i-l+1 > k){
                sum -= nums[l];
                mpp.put(nums[l], mpp.get(nums[l])-1);
                if(mpp.get(nums[l])<1) mpp.remove(nums[l]);
                l++;
            }
            if(i-l+1==k && mpp.size()==k){
                maxsum = Math.max(maxsum, sum);
            }
        }
        return maxsum;
    }
}

File: 2461-maximum-sum-of-distinct-subarrays-with-length-k/2461-maximum-sum-of-distinct-subarrays-with-length-k.cpp
class Solution {
public:
    long long maximumSubarraySum(vector<int>& nums, int k) {
        unordered_map<int, int> mpp;
        int l=0;
        long long sum = 0;
        long long maxsum = 0;

        for(int i=0;i<nums.size();i++){
            sum += nums[i];
            mpp[nums[i]]++;
            if(i-l+1 > k){
                sum -= nums[l];
                mpp[nums[l]]--;
                if(mpp[nums[l]]<1) mpp.erase(nums[l]);
                l++;
            }
            if(i-l+1==k && mpp.size()==k){
                maxsum = max(maxsum, sum);
            }
        }
        return maxsum;
    }
};